auth/auth.c

```diff
@@ -449,13 +449,17 @@ static void gip_auth_exchange_rsa(struct work_struct *work)
 					     work_exchange_rsa);
 	struct gip_auth_pkt_host_secret pkt = {};
 	u8 random[GIP_AUTH_RANDOM_LEN * 2];
-	u8 pms[GIP_AUTH_SECRET_LEN];
+	u8 *pms;
 	int err;
 
 	memcpy(random, auth->random_host, sizeof(auth->random_host));
 	memcpy(random + sizeof(auth->random_host), auth->random_client,
 	       sizeof(auth->random_client));
 
+	pms = kmalloc(GIP_AUTH_SECRET_LEN, GFP_KERNEL);
```

Comment:
Personally I would move this above the memcpy, there's no need to do this unless memory allocation succeeds.

Reply:
I wanted to restore it closest to the original commit, but you're right - it's better this way.